VertexKF[] vertices_kf
VertexMP[] vertices_mp
Edge[] edges
